lib/checkclass.cpp

```diff
@@ -541,6 +541,8 @@ void CheckClass::createUsageList(std::vector<Usage>& usageList, const Scope *sco
         usageList.push_back(Usage(&var));
     if (scope->definedType) {
         for (const Type::BaseInfo& baseInfo: scope->definedType->derivedFrom) {
+            if (scope->definedType == baseInfo.type) // don't crash on recursive templates
```

Comment:
are you sure that the recursion will happen immediately? I added a related fix but I added a vector of baseClasses.
205af353db3b1591427c4eaeeafbf2cdee8745dd

Reply:
Your fix solved the crash I saw also.. so I have replaced my fix.. 0338e99f84bceaab75cf9dd321d3e85a2935b0d3

Let's see how this work. Thanks!